### PR TITLE
Delete all old dag pages and redirect to grid view

### PR DESCRIPTION
**Pull request by bbovenzi**
All of the DAG pages are in React now, so we can delete the old FAB templates and redirect the urls to the grid view and appropriate tab. This removes extraneous vertical space on the page too.

Landing Times -> Run Duration
Task Duration -> Task Duration
Audit Log -> Audit Log
Calendar -> Calendar
Task Tries -> Grid (this never made sense as a line chart, instead try_number is displayed in more places)

Before:
Screenshot 2024-03-07 at 9 54 57 PM

After:
Screenshot 2024-03-07 at 9 54 38 PM

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by jscheffl — COMMENTED** [review]
Whohooo! Cool! Finally! (I just need to get used to the "missing top bar" - so much more space on the screen - but will take a moment until I take the new alternative buttons.

I'd approve but have a small request for consistency of beautification: There are 3,5 screens leftover in the old structure:
image

- Grid/Task Details --> More Details (still pointing to a legacy view, which actually is just a small leftover to migrate... wanted to contribute this but never came out of my backlog :-()
- Grid/Task Details --> Rendered Template (same here)
- Grid/Logs --> See more

If you click on any of these you end-up on the legacy view with the leftovers, still there is a top navigation, besides the browser back button there is no way to "officially" get back to Grid view. Also the legacy XCom is these.
image

Can you please add a button to get back to Grid view?
Would it make sense to drop the XCom view here as well? The new one is there anyway.
(Alternatively, wouls it be simple to migrate at elast the two "More Details/Rendered Template" also into React? Old Log view might stay there, at least I am a fan of the full screen option still from old times... sometimes logs in Grid are too small)

**Comment by bbovenzi** [comment]
> Alternatively, would it be simple to migrate at elast the two "More Details/Rendered Template" also into React? Old Log view might stay there, at least I am a fan of the full screen option still from old times... sometimes logs in Grid are too small)

Yes, I need to add rendered templates to React and then remove these pages too.
And I need to do more "Full screen" views.

File: airflow/www/views.py [hunk 1]
```diff
@@ -167,7 +164,7 @@
 logger = logging.getLogger(__name__)
 
 
-def sanitize_args(args: dict[str, str]) -> dict[str, str]:
+def sanitize_args(args: dict[str, Any]) -> dict[str, Any]:
```
Comment (uranusjr) [inline airflow/www/views.py:167]: Not related, but needed to make Mypy happy. See https://github.com/pallets/flask/issues/4979

**Comment by jscheffl** [comment]
Ready to test to get it into 2.9.0?

**Comment by bbovenzi** [comment]
> Ready to test to get it into 2.9.0?

Yep! Ready for review now.

**Review by jscheffl — APPROVED** [review]
I was hoping that the last two pages are also replaced with the last commit as notes in https://github.com/apache/airflow/pull/37988#pullrequestreview-1932895524 - but LGTM before having this waiting until 2.10 - leftovers can also be made in a follow-up PR...

**Comment by bbovenzi** [comment]
> I was hoping that the last two pages are also replaced with the last commit as notes in [#37988 (review)](https://github.com/apache/airflow/pull/37988#pullrequestreview-1932895524) - but LGTM before having this waiting until 2.10 - leftovers can also be made in a follow-up PR...

Which last two pages are you referring to? Finishing off the task specific ones like `rendered_templates`? That will be a follow-up PR.

**Comment by potiuk** [comment]
As discussed with @bbovenzi -> removal of "Required" dependencies is not as easy due to caching.

https://github.com/apache/airflow/pull/38312 adds mechanism to do so

**Comment by bbovenzi** [comment]
> As discussed with @bbovenzi -> removal of "Required" dependencies is not as easy due to caching.
> 
> #38312 adds mechanism to do so

Thanks! I'll do that in a follow-up PR

**Review by dstandish — APPROVED** [review]

**Comment by dstandish** [comment]
@bbovenzi i think this one really wants a news fragment.

File: airflow/www/views.py [hunk 2]
```diff
@@ -2836,110 +2833,11 @@ def legacy_calendar(self):
 
     @expose("/dags/<string:dag_id>/calendar")
     @auth.has_access_dag("GET", DagAccessEntity.RUN)
```
Comment (vincbeck) [inline airflow/www/views.py]: Since we are redirecting, no need to check the access.

```suggestion
```

File: airflow/www/views.py [hunk 3]
```diff
@@ -3080,149 +2978,9 @@ def legacy_duration(self):
 
     @expose("/dags/<string:dag_id>/duration")
     @auth.has_access_dag("GET", DagAccessEntity.TASK_INSTANCE)
```
Comment (vincbeck) [inline airflow/www/views.py]: Same

```suggestion
```

File: airflow/www/views.py [hunk 4]
```diff
@@ -3231,79 +2989,13 @@ def legacy_tries(self):
 
     @expose("/dags/<string:dag_id>/tries")
     @auth.has_access_dag("GET", DagAccessEntity.TASK_INSTANCE)
```
Comment (vincbeck) [inline airflow/www/views.py]: Same.

```suggestion
```

File: airflow/www/views.py [hunk 5]
```diff
@@ -3312,92 +3004,15 @@ def legacy_landing_times(self):
 
     @expose("/dags/<string:dag_id>/landing-times")
     @auth.has_access_dag("GET", DagAccessEntity.TASK_INSTANCE)
```
Comment (vincbeck) [inline airflow/www/views.py]:
```suggestion
```

File: airflow/www/views.py [hunk 6]
```diff
@@ -3862,64 +3477,22 @@ def legacy_audit_log(self):
 
     @expose("/dags/<string:dag_id>/audit_log")
     @auth.has_access_dag("GET", DagAccessEntity.AUDIT_LOG)
```
Comment (vincbeck) [inline airflow/www/views.py]:
```suggestion
```

**Review by vincbeck — COMMENTED** [review]
Love it!

**Review by ephraimbuddy — APPROVED** [review]

**Comment by jscheffl** [comment]
Finally.. WOHOOO! Thanks!

**Pull request merged**